SETUP/devex/README.md

```diff
@@ -0,0 +1,125 @@
+
+This environment uses the JSON-based development-only "forum" instead of a
+full phpBB forum system. This should be adequate for most things that aren't
+modifying the phpBB integration.
+
+### Set up
+
+First, install either [Docker Desktop](https://www.docker.com/products/docker-desktop/)
+-- free for personal use -- or [Rancher Desktop](https://rancherdesktop.io/).
+Note that if you use Rancher Desktop the commands are slightly different,
+consult their manual for details.
+
+_Note: All commands should be run relative to this directory._
+
+Start the docker containers:
+```bash
+docker compose up -d
+```
+
+On first run it will download some docker images and build the dp-dev image.
+When the containers start it will create an initial (empty) database and
+provide a web interface on port 8080.
+
+Install the PHP & JS packages:
+```bash
+docker compose exec dp-dev composer install
+docker compose exec dp-dev npm install
+```
+
+Run the configuration script:
+```bash
+docker compose exec dp-dev SETUP/configure SETUP/devex/dpdev-docker/configuration.sh .
+```
+
+Initialize the DB _structure_ with:
+```bash
+docker compose exec dp-dev bash -c "cd SETUP && php install_db.php"
+```
+
+Finally, initialize some of the "dynamic" directories that `configuration.sh`
+points to:
+```bash
+docker compose exec dp-dev SETUP/devex/dpdev-docker/init_dyndir.sh
+```
+
+Now you have a fully-installed and configured DProofreaders repository!
+You can access the site locally at http://localhost:8080/c/ and should be
+able to register a new user. See the [install guide](../INSTALL.md) for
+details on how to make that user an admin.
+
+**Note**: If you want to use the example data (see below) you must load that
+before registering new users or the example data will fail to load!
+
+When you're done you can bring down the containers:
+```bash
+docker compose down
+```
+
+### Loading example data
+
+It's kinda boring though. Let's populate it with some data like users, and a
+project to get started.
+
+This is, unfortunately a little fragile and the DB inserts may fail if the DB
+structure has changed.
+```bash
+docker compose exec dp-dev SETUP/devex/dpdev-docker/populate_data.sh
+```
+
+This creates 3 users with various permissions:
+* Site Admin
+  * username: `admin`
+  * password: `admin`
+* Project Manager
+  * username: `pm`
+  * password: `pm`
+* Proofreader
+  * username: `proofer`
+  * password: `proofer`
+
+A project is also loaded and available for proofreading. The project is managed
+by the `pm` user and the `proofer` user has already proofread a page in it.
+
+### Developing
+
+You can update the code directly from the repo and just refresh your browser.
+
+If you want to interact with the database you can do so through the dp-dev
+container:
+```bash
+docker compose exec dp-dev mysql -hmysql -udp_user -pdp_password dp_db
+```
+
+You can run php-cs-fixer and phpstan directly as well:
+```bash
+docker compose exec dp-dev ./vendor/bin/php-cs-fixer fix
```

Comment:
should we link to the https://github.com/DistributedProofreaders/dproofreaders/blob/master/SETUP/CODE_STYLE.md here for other commands that could be available?

Reply:
Good idea. I added the `npm run format` example as well a links to those.